master/buildbot/process/buildrequestdistributor.py

```diff
@@ -404,13 +406,20 @@ def xform(bldr):
 
         # sort the transformed list synchronously, comparing None to the end of
         # the list
-        def nonecmp(a, b):
+        def nonekey(a):
             if a[0] is None:
-                return 1
-            if b[0] is None:
-                return -1
-            return cmp(a, b)
-        xformed.sort(cmp=nonecmp)
+                b = list(a)
```

Comment:
I don't think this implementation is very complicated.
If no date is specified in an item, then the sort is supposed to place that item near the end of the list.  This behavior is preserved with my change.

Regarding the Activatestate code example you posted, cmp_to_key() does not exist
in Python 2.6, nor does it exist in the future module, so it cannot be used.